app/generate.js

```diff
@@ -1,106 +1,166 @@
+        }
+        return false;
+      });
+      page.sectionsHtml.push(section.html);
     }
-
-    retArray.push({link: page.link, output: pageTemplate( page )});
+    retArray.push({ link: page.link, output: pageTemplate(page) });
   }
-  //Writes out the html file and then saves it to a txt.
+  //  Writes out the html file and then saves it to a txt.
   return retArray;
 }
 
 function writePages(name, pageOutput) {
-  for (let i = 0; i < pageOutput.length; i++){
-    fs.writeFileSync('./dist/' + name + '/' + pageOutput[i].link + '.html', pageOutput[i].output);
+  for (let i = 0; i < pageOutput.length; i += 1) {
+    fs.writeFileSync(`./dist/${name}/${pageOutput[i].link}.html`, pageOutput[i].output);
   }
-
 }
 
-export default function generate() {
-  const orgData = getData("org");
-  const pageData = getData("pages");
-  const sectionsData = getData("sections");
-  const headerTemplate = template(fs.readFileSync('./templates/default/header.tpl.html', 'utf8'));
-  const footerTemplate = template(fs.readFileSync('./templates/default/footer.tpl.html', 'utf8'));
-  
-  orgData.header[0].html = headerTemplate({logo: orgData.header[0].logo});
-  //Puts the header from org through the headerTemplate and saves it in headerOuput.
-  orgData.footer[0].html = footerTemplate({contact: orgData.footer[0].contact});
-  //Does the same thing with header, but with footer.
-  createDistFolder(orgData.name)
-  //Makes a folder for the organization which we're going to save a file for.
-  //Is supposed to delete the file if it already exists.
-
-  const sectionOutput = generateSections(orgData.theme, sectionsData.sections);
-
-  const pageOutput = generatePages(pageData.pages,
-    orgData.footer,
-    orgData.header,
-    sectionOutput, );
+async function processOrg(orgId) {
+  const { Org: orgData } = await getGraphData('org', orgId);
+  const pageData = await getGraphData('pages', orgId);
+  const sectionData = await getGraphData('sections', orgId);
+  const imageData = await getGraphData('images', orgId);
+
+  const headerTemplate = template(fs.readFileSync(`./templates/default/header.tpl.html`, 'utf8'));
+  const footerTemplate = template(fs.readFileSync(`./templates/default/footer.tpl.html`, 'utf8'));
+
+  orgData.defaultHeader.html = headerTemplate({ logo: orgData.defaultHeader.logo.url });
+  //  Puts the header from org through the headerTemplate and saves it in headerOuput.
+  orgData.defaultFooter.html = footerTemplate({ contact: orgData.defaultFooter.email });
+  //  Does the same thing with header, but with footer.
+  createDistFolder(orgData.name);
+  //  Makes a folder for the organization which we're going to save a file for.
+  //  Is supposed to delete the file if it already exists.
+
+  const imageOutput = generateImages(orgData.theme, imageData.allImages);
+
+  const sectionOutput = generateSections(orgData.theme, sectionData.allSections, imageOutput);
+
+  const pageOutput = generatePages(orgData.theme, pageData.allPages,
```

Comment:
Since we have the other params for `generatePages` in the object, giving them nice names, please move `theme` and `pages` in there too. It'll look much cleaner that way!